### ISSUE TITLE
Smart wake up alarms

### ISSUE THREAD

**Issue by DanMcz** [AmazMod/AmazMod#804]
I use amazmod for a long time now, and I couldn't be happier with it. However, there's one feature I still miss from my watch: all the other my band and amazfit products that are simpler can be tweaked with an app to do smart alarm (you set a time period in which you want to wake up and it will wake you when you are in the lightest sleep phase)...

I know it's hard enough to fix the bugs, but I think it would be a really useful feature, which shouldn't be hard to implement (I guess) as amazmod service already stores heart rate info, can access to movement sensors and can rotate the vibromotor. Maybe you can also access the default sleep monitor app's data too?. 

I would happily test anything, and I hope it's not a problem that I wrote it here. (as a help: tools and amazfit, tools and miband, amazfit notification and tools apps are the ones that make every other watch capable of this, tools and miband even without connection)

Thank for all your work with amazmod!:)

**Comment by GreatApo** [comment]
We still don't have proper access to heart-rate and other data but there is definitely a way to do so (others have done it). Anyway, here are some more details about the requested feature:

Smart wake up alarm:
- set a Smart wake up alarm between specific time (eg. 07:30 and 08:00 AM)
- the alarm wakes you up sometime between 07:30 and 08:00 AM when you reach a state of light sleep (after a deep sleep)
- If you do not reach a state of light sleep during this period, the Smart wake up alarm wakes you up at 07:00 AM.

Basically, the app needs to look at your sleep data at 7:30am and if you have been in deep sleep and you are in light sleep, it wakes you up, else, loop with a time interval until you are in light sleep (or after been in light sleep for X min) or reach 8:00am.

**Comment by 1immortal** [comment]
sounds logical, I think that you meant to wake up at 08:00 AM.
is there a way to tap into the sleeping monitoring of the watch? all it does is HR measurements and the acceleration of the watch, right?

**Comment by DanMcz** [comment]
Yes, @GreatApo this is exactly what I meant, sorry for not making it clear directly.

I know that Notify and Fitness can get continuous heart rate info, but it probably get it from the amazfit app itself. 
It would be already great if you could do something like that app, and make it a plugin to a sleep related app (like "sleep as android", which can work now with every other amazfit product using Notify and Fitness as a bridge) but that would need continuous connection and could only use the phone as alarm and data analyzer, and I guess our watch can do way more.

If not a problem could you please tell me which watch provided data are reachable by amazmod service? 
If I know well: gyro, acceleration, wifi, pressure, and the single stored hr data, right? 
But what about saved locations, sleep data, alarms?
(with an app I could also reach common settings states and probably could edit them too) 
Because I may have some more suggestion, if those data are available that would be needed.

**Comment by fcaronte** [comment]
Old mi band has this feature from stock mifit app, I can't understand why huami not use this on his android device

**Comment by 1immortal** [comment]
Closing as this an as its a https://github.com/AmazMod/AmazMod/issues/75 duplicate
